src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
```

Comment:
Don't use class Objects for the `head` and `tail` references. Instead, specify the generic type like `Node<T>` to ensure type safety.